proof/refine/ARM/KHeap_R.thy

```diff
@@ -4288,4 +4288,64 @@ global_interpretation set_simple_ko: typ_at_pres "set_simple_ko C ptr ep"
 global_interpretation update_sk_obj_ref: typ_at_pres "update_sk_obj_ref C update ref new"
   unfolding typ_at_pres_def by wpsimp
 
+
+lemma getReprogramTimer_corres:
+  "corres (=) \<top> \<top> (gets reprogram_timer) getReprogramTimer"
+  by (clarsimp simp: getReprogramTimer_def state_relation_def)
+
+lemma setDomainTime_corres:
+  "dt = dt' \<Longrightarrow>
+  corres dc \<top> \<top> (modify (domain_time_update (\<lambda>_. dt))) (setDomainTime dt')"
+  apply (clarsimp simp: setDomainTime_def, rule corres_modify)
+  by (clarsimp simp: state_relation_def swp_def)
+
+lemma setConsumedTime_corres:
+  "ct = ct' \<Longrightarrow>
+  corres dc \<top> \<top> (modify (consumed_time_update (\<lambda>_. ct))) (setConsumedTime ct')"
+  apply (clarsimp simp: setConsumedTime_def, rule corres_modify)
+  by (clarsimp simp: state_relation_def swp_def)
+
+lemma setCurSc_corres:
+  "sc = sc' \<Longrightarrow>
+   corres dc \<top> \<top> (modify (cur_sc_update (\<lambda>_. sc))) (setCurSc sc')"
+  apply (clarsimp simp: setCurSc_def, rule corres_modify)
+  by (clarsimp simp: state_relation_def swp_def)
+
+lemma machine_op_lift_corres_trivial:
+  "f = f' \<Longrightarrow> corres_underlying Id False True dc \<top> \<top> (machine_op_lift f) (machine_op_lift f')"
```

Comment:
This seems like a little bit of a strange lemma to me. It's useful, but couldn't it follow more directly from `corres_underlying_trivial`?

Reply:
Oops! Good point!